anodizer/anodizer-core/src/main/kotlin/io/github/amzn/anodizer/core/File.kt

```diff
@@ -4,13 +4,20 @@ import java.io.ByteArrayInputStream
 import java.io.ByteArrayOutputStream
 import java.io.InputStream
 import java.io.PrintStream
+import java.util.jar.JarFile
 
 /**
  * Rudimentary filesystem backed by ByteArrayInputStream and ByteArrayOutputStream.
  *
  * Q: Why?
  * A: A generator produces a file tree which callers consume.
  *
+ * Q: More why?
+ * A: We can re-create a file hierarchy from the resources in a JAR.
+ *
+ * Q: What is wrong with this?
+ * A: For starters, there are no assertions on duplicate entities.
+ *
  * TODO tree pretty-print helper.
  */
 public class File private constructor(
```

Comment:
So. Trees of `File` are generation target for codegen. And this change adds `resource` to load and 'copy' files directly into the output.

Reply:
Yes. A target produces a File (possibly a tree of dirs and files) which is a simple in-memory filesystem which a consumer can then materialize. The smithy integration walks this tree and creates the actual files through the smithy FileManifest abstraction. This ensures targets can generate arbitrarily complex output (more than just one stream of bytes) and Smithy has control of how it is actually written.